certbot/interfaces.py

```diff
@@ -238,6 +238,12 @@ class IConfig(zope.interface.Interface):
         "This only affects the port Certbot listens on. "
         "A conforming ACME server will still attempt to connect on port 80.")
 
+    http01_use_tls = zope.interface.Attribute(
+        "Runs the standalone server with a self signed certificate. "
+        "Useful for when you need to validate an http-01 challenge with "
+        "an https endpoint or if you have an upstream proxy redirecting "
+        "all http traffic to https.")
```

Comment:
Right, and if I have a proxy that redirects all http to https traffic (which is not un-common), then there is no entry point for upstream https servers running the acme client to validate an http-01 challenge, unless they are validating the challenge on https. Since I'm also doing tls-sni based routing (and can't use tls-sni-01), I've expressed this on the acme mailing list - which I believe you gave some feedback on. There is no way for me to use the acme protocol to install certificates.. :face_with_thermometer: